src/services/completions.ts

```diff
@@ -52,6 +52,11 @@ namespace ts.Completions {
         return !!(origin.kind & SymbolOriginInfoKind.Nullable);
     }
 
+    interface UniqueNameSet {
+        add(name: string): void;

```

Comment:
Some collection types return a boolean from insertion indicating whether the collection was updated.  This has the advantage that you can add-if-not-present in a single call.

Reply:
The purpose of this type is just to obscure the value type and prevent consumers from testing the result of the underlying Map’s `get` method for truthiness, since `false` indicates something very different from `undefined`. I was actually hoping to just _use_ the Map object and obscure its value type with this interface, but I couldn’t do that because the `set` method needs two arguments, one of which is a value.